extras/docker/alpine3.18/Dockerfile

```
@@ -0,0 +1,48 @@
+FROM alpine:3.18 AS builder
+
+ARG MONGOCXX_VERSION
+ARG MONGOC_VERSION=1.24.2
+ARG MONGOCRYPT_BRANCH=r1.8
+
+WORKDIR /build
+
+RUN apk add --no-cache \
+    alpine-sdk \
+    cmake \
+    openssl-dev \
+    python3
+
+RUN mkdir /install
+
+RUN git clone https://github.com/mongodb/libmongocrypt.git --branch ${MONGOCRYPT_BRANCH} && \
+    cd libmongocrypt && \
+    mkdir cmake-build && \
+    cd cmake-build && \
+    cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/install && \
+    make -j $(nproc) && \
+    make install -j $(nproc)
+
+RUN wget https://github.com/mongodb/mongo-c-driver/releases/download/${MONGOC_VERSION}/mongo-c-driver-${MONGOC_VERSION}.tar.gz && \
```

As of CDRIVER-4640, there is no longer a source archive produced in the C driver.

Suggest using the GitHub produced source archive, so this command will work with the eventual C driver 1.25.0 release and newer.
```suggestion
RUN wget https://github.com/mongodb/mongo-c-driver/archive/refs/tags/${MONGOC_VERSION}.tar.gz && \
```

If this change is applied, also add `BUILD_VERSION` in the cmake configuration command `-D BUILD_VERSION=${MONGOC_VERSION}`. There is no VERSION_CURRENT file present in the GitHub produced source archive. Without specifying `BUILD_VERSION` I expect the calculated version will be 0.0.0.

Suggestion also applies to other Dockerfiles.

---

I've updated all of the dependencies (i.e., mongoc, mongocxx, and libmongocrypt) to use this archive path.